cli/src/ui.rs

```diff
@@ -140,6 +81,26 @@ impl UiOutput {
         Ok(UiOutput::Paged { child, child_stdin })
     }
 
+    fn new_builtin_paged() -> streampager::Result<UiOutput> {
+        let mut pager = streampager::Pager::new_using_stdio()?;
+        // TODO: should we set the interface mode to be "less -FRX" like?
+        // It will override the user-configured values.
```

Comment:
Could you elaborate on "It will override the user-configured values."? Can Streampager be configured by the user outside `jj`? 

I think we should default to the ["Hybrid" behavior](https://docs.rs/sapling-streampager/latest/streampager/config/enum.InterfaceMode.html#variant.Hybrid), at least if the user hasn't configured anything.

Reply:
> Streampager be configured by the user outside jj?

Yes. https://github.com/markbt/streampager?tab=readme-ov-file#configuration

And setting non-default value here means user config wouldn't be respected afaik. I don't have an opinion about the default behavior, but it should be configurable somehow.

Reply:
> Yes. https://github.com/markbt/streampager?tab=readme-ov-file#configuration

Thank you!

> And setting non-default value here means user config wouldn't be respected afaik. 

Do you think it'd make sense to check whether Streampager's config file exists, and use our default if it doesn't?

Reply:
Actually, I don't think `jj`'s streampager respects `$CONFIG_DIR/streampager/streampager.toml`. I have both `~/.config/streampager/streampager.toml` and `~/Library/Application Support/streampager/streampager.toml`, and it doesn't seem to be respected. Is it for you?

Sapling also seems to duplicate streampager config. I put an excerpt from `sl help config` in https://gist.github.com/ilyagr/8760765932c2b6f4b70ea87a0bf5c804 .

Reply:
As a complete aside, the `pager.ignore` setting from that Gist is an interesting idea.

Reply:
> Do you think it'd make sense to check whether Streampager's config file exists, and use our default if it doesn't?

Sounds like a bad idea. Maybe better to add our config section for the builtin pager.

> Actually, I don't think jj's streampager respects $CONFIG_DIR/streampager/streampager.toml

`strace` shows the file is read. Appears that `interface_mode` isn't parsed however, maybe because of serde or toml API change?

Reply:
Yes, it's supposed to be read, and some environment variables are supposed to affect it as well.

https://docs.rs/crate/sapling-streampager/0.10.3/source/src/config.rs#201
https://docs.rs/crate/sapling-streampager/0.10.3/source/src/pager.rs#108

Clearly, it's buggy.

Unfortunately, Streampager does not seem to have a `new_with_config` function. I wonder whether we could disable all of that streampager native config by setting `pager.config` ourselves after `pager` is created. Then, we could have our own config for the builtin pager, just as you said.

This would mean we'd be uselessly reading that file. We could contribute some form of `new_with_config` to streampager if it makes sense for them and their API.

Reply:
So, the best option might be to copy Sapling: set the three options we can in `jj` config, overriding streampager's defaults with our own. Without any config, Sapling defaults to the Hybrid mode, which is the default for the Sapling config setting, but not the default for streampager's.

Reply:
I'll do that.